pkg/kudoctl/cmd/diagnostics/resource_funcs.go

```diff
@@ -0,0 +1,152 @@
+package diagnostics
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"reflect"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/env"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/kudoinit"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/kudo"
+	kudoutil "github.com/kudobuilder/kudo/pkg/util/kudo"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+)
+
+// resourceFuncsConfig - a wrapper for Kube and Kudo clients and common invocation parameters
+// for loading Kube and Kudo resources
+type resourceFuncsConfig struct {
+	c           *kudo.Client
+	ns          string
+	instanceObj *v1beta1.Instance
+	opts        metav1.ListOptions
+	logOpts     corev1.PodLogOptions
+}
+
+// newInstanceResources is a configuration for instance-related resources
+func newInstanceResources(instanceName string, options *Options, c *kudo.Client, s *env.Settings) (*resourceFuncsConfig, error) {
+	instance, err := c.GetInstance(instanceName, s.Namespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get instance %s/%s: %v", s.Namespace, instanceName, err)
+	}
+	if instance == nil {
+		return nil, fmt.Errorf("instance %s/%s not found", s.Namespace, instanceName)
+	}
+	return &resourceFuncsConfig{
+		c:           c,
+		ns:          s.Namespace,
+		instanceObj: instance,
+		opts:        metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s", kudoutil.OperatorLabel, instance.Labels[kudoutil.OperatorLabel])},
+		logOpts:     corev1.PodLogOptions{SinceSeconds: options.LogSince},
+	}, nil
+}
+
+// newKudoResources is a configuration for Kudo controller related resources
+// panics if used to load Kudo CRDs (e.g. instance etc.)
+func newKudoResources(options *Options, c *kudo.Client) (*resourceFuncsConfig, error) {
+	opts := metav1.ListOptions{LabelSelector: fmt.Sprintf("app=%s", kudoinit.DefaultKudoLabel)}
+	ns, err := c.CoreV1().Namespaces().List(opts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get kudo system namespace: %v", err)
+	}
+	if ns == nil || len(ns.Items) == 0 {
+		return nil, fmt.Errorf("kudo system namespace not found")
+	}
+	return &resourceFuncsConfig{
+		c:       c,
+		ns:      ns.Items[0].Name,
+		opts:    opts,
+		logOpts: corev1.PodLogOptions{SinceSeconds: options.LogSince},
+	}, nil
+}
+
+// object implements runtime.object and
+// metav1.object interfaces.
+// copied from K8 internal type metaRuntimeInterface
+type object interface {
+	runtime.Object
+	metav1.Object
+}
```

Comment:
Erks. Do we really need this? I was already wondering above how it was possible to cast `obj.(object)` when the comment talked about `metav1.Object`.

Reply:
I just looked it up in the code, we *can* delete it. It's only used in the `printSingleObject` and it's used to call `getName` on the `metav1.Object`. I'll comment up there

Reply:
done